src/lib.rs

```diff
@@ -140,24 +140,24 @@ fn json_del(ctx: &Context, args: Vec<String>) -> RedisResult {
 ///
 /// JSON.SET <key> <path> <json> [NX | XX | FORMAT <format>]
 ///
-fn json_set(ctx: &Context, args: Vec<String>) -> RedisResult {
+fn json_set(ctx: &Context, args: Vec<RedisString>) -> RedisResult {
     let mut args = args.into_iter().skip(1);
 
-    let key = args.next_string()?;
+    let key = args.next_redis_string()?;
     let path = backwards_compat_path(args.next_string()?);
     let value = args.next_string()?;
 
     let mut format = Format::JSON;
     let mut set_option = SetOptions::None;
 
     while let Some(s) = args.next() {
-        match s.to_uppercase().as_str() {
-            "NX" => set_option = SetOptions::NotExists,
-            "XX" => set_option = SetOptions::AlreadyExists,
-            "FORMAT" => {
+        match s.try_as_str()? {
+            arg if arg.eq_ignore_ascii_case("NX") => set_option = SetOptions::NotExists,
+            arg if arg.eq_ignore_ascii_case("XX") => set_option = SetOptions::AlreadyExists,
+            arg if arg.eq_ignore_ascii_case("FORMAT") => {
                 format = Format::from_str(args.next_string()?.as_str())?;
             }
-            _ => break,
+            _ => break, // TODO shoudl return an error??
```

Comment:
It probably should.

Reply:
Before commit 89b8a957e27e8f9a8e46b354f930937c6c331522 there seems to have been an error report `Err(RedisError::Str("ERR syntax error"))`